### PR TITLE
chore: cleanup and refactor sign code

### PR DESCRIPTION
**Pull request by derklaro**
Cleans up the sign system code by a lot to bring more consistency and easier readability. This should also improve the performance of the sign system, as for example copying the sign location to a platform location is no longer needed when updating the current shown lines of a sign.

### CODE REVIEW

File: modules/signs/src/main/java/eu/cloudnetservice/modules/signs/Sign.java [hunk 1]
```diff
@@ -34,7 +30,7 @@
  */
 @ToString
 @EqualsAndHashCode(onlyExplicitlyIncluded = true)
-public class Sign implements Comparable<Sign> {
+public class Sign {
 
   public static final Type COLLECTION_TYPE = new TypeToken<Collection<Sign>>() {
```
Comment (TheMeinerLP) [inline modules/signs/src/main/java/eu/cloudnetservice/modules/signs/Sign.java:35]: Can you change this to better way ?

File: modules/signs/src/main/java/eu/cloudnetservice/modules/signs/platform/sponge/SpongePlatformSign.java [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+    }
+
+    // get the block at the given location
+    var entity = location.blockEntity().orElse(null);
+    if (entity instanceof org.spongepowered.api.block.entity.Sign sign) {
+      // set the glowing status if needed
+      sign.glowingText().set(layout.glowingColor() != null);
+
+      // set the sign lines
+      for (int i = 0; i < Math.min(4, layout.lines().size()); i++) {
```
Comment (TheMeinerLP) [inline modules/signs/src/main/java/eu/cloudnetservice/modules/signs/platform/sponge/SpongePlatformSign.java:85]: What happend if i set less as 4 lines in the config ?

Comment (derklaro) [reply modules/signs/src/main/java/eu/cloudnetservice/modules/signs/platform/sponge/SpongePlatformSign.java:85]: Well, Math.min returns the smaller of a and b - if you define more than 4 lines only 4 will be used, if you define less then that amount will be used instead.

Comment (TheMeinerLP) [reply modules/signs/src/main/java/eu/cloudnetservice/modules/signs/platform/sponge/SpongePlatformSign.java:85]: Okay perfect 👍🏻

File: modules/signs/src/main/java/eu/cloudnetservice/modules/signs/Sign.java (hunk 1 above)
Comment (derklaro) [reply modules/signs/src/main/java/eu/cloudnetservice/modules/signs/Sign.java:35]: Not without breaking compatibility with older spigot versions

Comment (TheMeinerLP) [reply modules/signs/src/main/java/eu/cloudnetservice/modules/signs/Sign.java:35]: Can be make a issue.

**Review by 0utplay — APPROVED** [review]
lgtm

**Pull request merged**